src/main/java/org/jenkinsci/plugins/parallel_test_executor/ParallelTestExecutor.java

```diff
@@ -155,14 +161,60 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListen
         return true;
     }
 
+    public static Map<String, TestClass>  findTestResultsInDirectory(Run<?,?> build, TaskListener listener, @CheckForNull FilePath workspace){
+        if(workspace==null){
+            return Collections.emptyMap();
+        }
+        String[] tests = null;
+        Map<String, TestClass> data = new TreeMap<String, TestClass>();
+        final String baseDir = workspace.getRemote();
+        String separator = null;
+        final List<String> testFilesExpression = new ArrayList<String>();
+        testFilesExpression.add("**/src/test/java/**/Test*.java");
+        testFilesExpression.add("**/src/test/java/**/*Test.java");
+        testFilesExpression.add("**/src/test/java/**/*Tests.java");
+        testFilesExpression.add("**/src/test/java/**/*TestCase.java");
+        try {
+            separator = workspace.act(new MasterToSlaveCallable<String, Throwable>() {
```

Comment:
you could have just used a regular expression and avoid a callable.  `[\/]`  (would obviously need the correct escaping applied)

Reply:
I would let it be as it is. There is a big probability that this part will be changed when it was used on Windows according to reported issues.